app/scripts/modules/core/src/pipeline/config/stages/wercker/werckerStage.js

```diff
@@ -0,0 +1,170 @@
+        return lines + (stage.masterStage.context.buildInfo.testResults || []).length;
+      },
+      defaultTimeoutMs: 2 * 60 * 60 * 1000, // 2 hours
+      validators: [{ type: 'requiredField', fieldName: 'job' }],
+      strategy: true,
+    });
+  })
+  .controller('WerckerStageCtrl', function($scope, stage) {
+    $scope.stage = stage;
+    $scope.stage.failPipeline = $scope.stage.failPipeline === undefined ? true : $scope.stage.failPipeline;
+    $scope.stage.continuePipeline = $scope.stage.continuePipeline === undefined ? false : $scope.stage.continuePipeline;
+
+    $scope.viewState = {
+      mastersLoaded: false,
+      mastersRefreshing: false,
+      jobsLoaded: false,
+      jobsRefreshing: false,
+      failureOption: 'fail',
+      markUnstableAsSuccessful: !!stage.markUnstableAsSuccessful,
+      waitForCompletion: stage.waitForCompletion || stage.waitForCompletion === undefined,
+    };
+
+    // Using viewState to avoid marking pipeline as dirty if field is not set
+    this.markUnstableChanged = () => (stage.markUnstableAsSuccessful = $scope.viewState.markUnstableAsSuccessful);
+
+    this.waitForCompletionChanged = () => (stage.waitForCompletion = $scope.viewState.waitForCompletion);
+
+    function initializeMasters() {
+      IgorService.listMasters(BuildServiceType.Wercker).then(function(masters) {
+        $scope.masters = masters;
+        $scope.viewState.mastersLoaded = true;
+        $scope.viewState.mastersRefreshing = false;
+      });
+    }
+
+    this.refreshMasters = function() {
+      $scope.viewState.mastersRefreshing = true;
+      initializeMasters();
+    };
+
+    this.refreshApps = function() {
+      $scope.viewState.appsRefreshing = true;
+      updateAppsList();
+    };
+
+    function updateAppsList() {
+      if ($scope.stage && $scope.stage.master) {
+        $scope.viewState.appsLoaded = false;
+        $scope.apps = [];
+        IgorService.listJobsForMaster($scope.stage.master).then(function(jobs) {
+          var apps = {};
+          jobs.forEach(function(app) {
+              if(!app.startsWith('pipeline')) {
+                  var orgApp = app.substring(app.indexOf('/') + 1, app.lastIndexOf('/'));
+                  apps[orgApp] = orgApp;
+              }
+          });
+          $scope.apps = Object.keys(apps);
+          $scope.jobs = jobs;
+          $scope.viewState.appsLoaded = true;
+          $scope.viewState.appsRefreshing = false;
+          if (jobs.length && !$scope.apps.includes($scope.stage.app)) {
+            $scope.stage.app = '';
+          }
+        });
+      }
+    }
+
+    this.refreshJobs = function() {
+      $scope.viewState.jobsRefreshing = true;
+      updateJobsList();
+    };
+
+    function updateJobsList() {
+      if ($scope.stage && $scope.stage.app) {
+        $scope.pipelines = [];
+        $scope.viewState.jobsLoaded = true;
+        $scope.viewState.jobsRefreshing = false;
+        if ($scope.jobs) {
+            var pipelines = {};
```

Comment:
Prefer `const`/`let` over var declarations.

Reply:
fixed all the var to let